src/api/config/clock.rb

```diff
@@ -48,9 +50,17 @@ module Clockwork
     FullTextSearch.new.delay.index_and_start
   end
 
-  every(1.day, 'refresh dirties') do
-    # inject a delayed job for every dirty project
-    BackendPackage.refresh_dirty
+  every(1.hour, 'refresh dirties', thread: true) do
+    unless update_backend_infos && update_backend_infos.running?
+      update_backend_infos = Daemons.call do
+        loop do
+          Package.dirty_backend_packages.each do |pkg|
```

Comment:
Depending on the amount of packages we might fetch here (>1000 ?) it might make sense to fetch them in batches -> http://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each

Not sure if that works with the new implementation. With scopes it would.